### ISSUE TITLE
IDE0068 Triggers invalidly for awaited disposables already in a using

### ISSUE THREAD

**Issue by NickCraver** [dotnet/roslyn#37065]
**Version Used**: VS 16.2 Preview 3

**Steps to Reproduce**:
This code will trigger IDE0068:
```c#
using (var conn = await GetConnectionAsync())
{
    return await conn.ExecuteAsync("DBCC FREEPROCCACHE (@planHandle);", new { planHandle });
}
```
Where `GetConnectionAsync()` is a `Task<DbConnection>`, like so:
```c#
protected Task<DbConnection> GetConnectionAsync() =>
  Connection.GetOpenAsync(ConnectionString);
```
And at the base (simplified):
```c#
public static async Task<DbConnection> GetOpenAsync(string connectionString) { }
```
Inside, this opens and returns a SQL connection via `.OpenAsync()`.

**Expected Behavior**: No warning when the usage *is* disposed, at least in the first `using` case. The 2 base methods: yeah okay, we can attribute them.

**Actual Behavior**: IDE0068 is issued for all 3 methods above.

100% of the cases we're seeing this is in with `await`. I tried to minimally repro this but could not, so I reverted my suppressions and made a repro available in Opserver. If you pull [this commit](https://github.com/opserver/Opserver/commit/c859a2d61226c9e9a475eac4d3d93f5006d25e08) it should give a slew of IDE0068 warnings that haven't been refactored around yet, most around paths in the connection bits mentioned above.

Getting a connection (asynchronously) and using it elsewhere is a *very* common pattern. Can we please handle the `await` case, or perhaps disable IDE0068 for the 16.2 GA release if it can't be resolved in time?

**Comment by mavasani** [comment]
I have a minimal repro:

```csharp
using System.IO;
using System.Threading.Tasks;

public class C
{
    public bool Flag;
    public async Task M()
    {
        using (var s = await GetStreamAsync().ConfigureAwait(false))
        {
        }
    }
    private async Task<MemoryStream> GetStreamAsync()
    {
        var x = new MemoryStream();
        if (Flag) x.Dispose(); // Commenting out this code leads to no IDE0068
        return x;
    }
}
```
